client/my-sites/controller.js

```diff
@@ -547,21 +547,26 @@ export function navigation( context, next ) {
  * @param {Function} next -- Call next middleware in chain
  */
 export function sites( context, next ) {
-	if ( context.query.verified === '1' ) {
-		context.store.dispatch(
-			successNotice(
-				i18n.translate(
-					"Email verified! Now that you've confirmed your email address you can publish posts on your blog."
+	const isAgencyDashboardEnabled = config.isEnabled( 'jetpack/agency-dashboard' );
+	if ( isAgencyDashboardEnabled ) {
+		page.redirect( '/dashboard' );
+	} else {
```

Comment:
For safety, I recommend adding a `return` statement inside this block -- [`page.direct` is an asynchronous call](https://github.com/visionmedia/page.js/blob/master/page.js#L660), so it's possible other routing code will execute after we decide to redirect. We can also then unwrap the enclosing `else` block immediately after:

```suggestion
	if ( isAgencyDashboardEnabled ) {
		page.redirect( '/dashboard' );
		return;
```

Reply:
Makes sense, will add the suggested change.